extensions/ipynb/src/notebookImagePaste.ts

```diff
@@ -0,0 +1,146 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+class CopyPasteEditProvider implements vscode.DocumentPasteEditProvider {
+
+	async provideDocumentPasteEdits(
+		_document: vscode.TextDocument,
+		_ranges: readonly vscode.Range[],
+		dataTransfer: vscode.DataTransfer,
+		_token: vscode.CancellationToken
+	): Promise<vscode.DocumentPasteEdit | undefined> {
+
+		// get b64 data from paste
+		// TODO: dataTransfer.get() limits to one image pasted
+		const dataItem = dataTransfer.get('image/png');
+		if (!dataItem) {
+			return undefined;
+		}
+		const fileDataAsUint8 = await dataItem.asFile()?.data();
+		if (!fileDataAsUint8) {
+			return undefined;
+		}
+		const b64string = encodeBase64(fileDataAsUint8);
+
+		// get filename data from paste
+		let pasteFilename = dataItem.asFile()?.name;
+		if (!pasteFilename) {
+			return undefined;
+		}
+		const separatorIndex = pasteFilename?.lastIndexOf('.');
+		const filename = pasteFilename?.slice(0, separatorIndex);
+		const filetype = pasteFilename?.slice(separatorIndex);
+		if (!filename || !filetype) {
+			return undefined;
+		}
+
+		// get notebook cell data
+		let currentCell;
+		for (const notebook of vscode.workspace.notebookDocuments) {
+			if (notebook.uri.path === _document.uri.path) {
+				for (const cell of notebook.getCells()) {
+					if (cell.document === _document) {
+						currentCell = cell;
+						break;
+					}
+				}
+
+			}
+		}
+		if (!currentCell) {
+			return undefined;
+		}
+		const initialAttachments = currentCell.metadata?.custom?.attachments;
+		const initialize = !initialAttachments ? true : false;
```

Comment:
Instead having to use `initialize`, can you just modify `currentCell.metadata` (potentially after cloning it)